javascript/javascript1/week2/student-manager.js

```diff
@@ -0,0 +1,58 @@
+
+const class07Students = [];
+function addStudentToClass(studentName) {
+    if (studentName) {
+        if (class07Students.length < 6) {
+            if (class07Students.includes(studentName)) {
+                console.log(`Student ${studentName} is already in the class`)
+                }
+            else {
+                return class07Students.push(studentName)
+                } 
+        }else if (studentName === 'Queen'){
+            return class07Students.push('Queen')
+        }else {
+            console.log('Cannot add more students to class 07') 
+            }
+        }
+   
+}
+
+function getNumberOfStudents() {
+  return class07Students.length
+}
+
+
+addStudentToClass('Benjamin')
+console.log(class07Students)
+console.log(getNumberOfStudents())
+addStudentToClass('')
+console.log(class07Students)
+console.log(getNumberOfStudents())
+addStudentToClass('Kiarash')
+console.log(class07Students)
+console.log(getNumberOfStudents())
+addStudentToClass('Jalal')
+console.log(class07Students)
+console.log(getNumberOfStudents())
+addStudentToClass('Zahra')
+console.log(class07Students)
+console.log(getNumberOfStudents())
+addStudentToClass('Benjamin')
+console.log(class07Students)
+console.log(getNumberOfStudents())
+addStudentToClass('Maxine')
+console.log(class07Students)
+console.log(getNumberOfStudents())
+addStudentToClass('Helle')
+console.log(class07Students)
+console.log(getNumberOfStudents())
+addStudentToClass('Rune')
+console.log(class07Students)
+console.log(getNumberOfStudents())
+addStudentToClass('Mary')
+console.log(class07Students)
+console.log(getNumberOfStudents())
+addStudentToClass('Queen')
+console.log(class07Students)
+console.log(getNumberOfStudents())
```

Comment:
Try to add `Queen` multiple times AFTER the class has filled. :wink:

But before you try to fix the bug I would recommend that you try to simplify the implementation with less nesting of `if` conditions.